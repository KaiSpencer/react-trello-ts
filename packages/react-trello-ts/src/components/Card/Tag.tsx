import React, { CSSProperties, FC, PropsWithChildren } from "react";
import { TagSpan } from "../../styles/Base";

export interface TagProps {
	title: string;
	color?: string;
	bgcolor?: string;
	tagStyle?: CSSProperties;
}
export const Tag: FC<PropsWithChildren<TagProps>> = ({
	title,
	color,
	bgcolor,
	tagStyle,
	...otherProps
}) => {
	const style = {
		color: color || "white",
		backgroundColor: bgcolor || "orange",
		...tagStyle,
	};
	return (
		<TagSpan style={style} {...otherProps}>
			{title}
		</TagSpan>
	);
};
