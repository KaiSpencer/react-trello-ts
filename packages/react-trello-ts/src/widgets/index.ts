export { DeleteButton } from "./DeleteButton";
export { EditableLabel } from "./EditableLabel";
export { InlineInput } from "./InlineInput";
