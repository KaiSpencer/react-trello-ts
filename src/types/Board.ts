import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
