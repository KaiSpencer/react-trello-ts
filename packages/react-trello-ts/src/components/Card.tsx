import React, {
	CSSProperties,
	EventHandler,
	FC,
	MouseEvent,
	PropsWithChildren,
} from "react";
import {
	MovableCardWrapper,
	CardHeader,
	CardRightContent,
	CardTitle,
	Detail,
	Footer,
} from "../styles/Base";
import { InlineInput } from "../widgets/InlineInput";
import { Tag, TagProps } from "./Card/Tag";
import { DeleteButton } from "../widgets/DeleteButton";
import { createTranslate } from "..";
import { Card as ICard } from "../types/Board";
import { StyledComponent } from "styled-components";

/**
 * Card component type
 *
 * Pass in a type to the optional generic to add custom properties to the card
 *
 * @example
 *
 * type CustomCardProps = {
 * 	dueOn: string;
 * }
 *
 * const CustomCard: CardComponent<CustomCardProps> = ({ dueOn, ...props }) => {
 * 	return (
 * 		<Card {...props}>
 * 			<Detail>{dueOn}</Detail>
 * 		</Card>
 * 	)
 * }
 */
export type CardComponent<TCustomCardProps extends {} = {}> = FC<
	PropsWithChildren<CardProps & TCustomCardProps>
>;

export type CardProps = {
	showDeleteButton?: boolean;
	onDelete?: () => void;
	onClick?: (e) => void;
	onChange?: (card: ICard) => void;
	style?: CSSProperties;
	tagStyle?: CSSProperties;
	className?: string;
	id: string;
	index: number;
	title?: string;
	label?: string;
	description?: string;
	tags?: TagProps[];
	cardDraggable?: boolean;
	editable?: boolean;
	t: typeof createTranslate;
};

export const Card: CardComponent = ({
	onDelete,
	onChange,
	id,
	onClick,
	style,
	className,
	description,
	label,
	t,
	tags,
	title,
	cardDraggable,
	editable,
	showDeleteButton,
	tagStyle,
}) => {
	const _onDelete = (
		e:
			| React.MouseEvent<HTMLDivElement>
			| React.MouseEvent<StyledComponent<"div", any>>,
	) => {
		onDelete();
		e.stopPropagation();
	};
	const updateCard = (card: Partial<ICard>) => {
		onChange({ ...card, id });
	};

	return (
		<MovableCardWrapper
			data-id={id}
			onClick={onClick}
			style={style}
			className={className}
		>
			<CardHeader>
				<CardTitle draggable={cardDraggable}>
					{editable ? (
						<InlineInput
							value={title}
							border={true}
							placeholder={t("placeholder.title") as unknown as string}
							resize="vertical"
							onSave={(value: ICard["title"]) => updateCard({ title: value })}
						/>
					) : (
						title
					)}
				</CardTitle>
				<CardRightContent>
					{editable ? (
						<InlineInput
							value={label}
							border={true}
							placeholder={t("placeholder.label") as unknown as string}
							resize="vertical"
							onSave={(value: ICard["label"]) => updateCard({ label: value })}
						/>
					) : (
						label
					)}
				</CardRightContent>
				{showDeleteButton && <DeleteButton onClick={_onDelete} />}
			</CardHeader>
			<Detail>
				{editable ? (
					<InlineInput
						value={description}
						border={true}
						placeholder={t("placeholder.description") as unknown as string}
						resize="vertical"
						onSave={(value: ICard["description"]) =>
							updateCard({ description: value })
						}
					/>
				) : (
					description
				)}
			</Detail>
			{tags && tags.length > 0 && (
				<Footer>
					{tags.map((tag) => (
						<Tag key={tag.title} {...tag} tagStyle={tagStyle} />
					))}
				</Footer>
			)}
		</MovableCardWrapper>
	);
};
