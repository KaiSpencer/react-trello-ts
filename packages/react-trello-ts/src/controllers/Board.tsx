import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
