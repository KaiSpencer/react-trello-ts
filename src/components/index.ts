import {LaneHeader} from './Lane/LaneHeader'
import {LaneFooter} from './Lane/LaneFooter'
import {Card} from './Card'
import {Loader} from './Loader'
import {NewLaneForm} from './NewLaneForm'
import {NewCardForm} from './NewCardForm'
import {AddCardLink} from './AddCardLink'
import {NewLaneSection} from './NewLaneSection'
import {BoardWrapper, GlobalStyle, ScrollableLane, Section} from 'rt/styles/Base'

export {
  GlobalStyle,
  BoardWrapper,
  Loader,
  ScrollableLane,
  LaneHeader,
  LaneFooter,
  Section,
  NewLaneForm,
  NewLaneSection,
  NewCardForm,
  Card,
  AddCardLink
}
