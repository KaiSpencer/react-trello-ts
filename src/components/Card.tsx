import React, {CSSProperties, EventHandler, FC, MouseEvent, PropsWithChildren} from 'react'
import {MovableCardWrapper, CardHeader, CardRightContent, CardTitle, Detail, Footer} from 'rt/styles/Base'
import {InlineInput} from 'rt/widgets/InlineInput'
import {Tag, TagProps} from './Card/Tag'
import {DeleteButton} from 'rt/widgets/DeleteButton'
import {createTranslate} from '..'
import {Card as ICard} from 'rt/types/Board'
import {StyledComponent} from 'styled-components'

export type CardProps = {
  showDeleteButton?: boolean
  onDelete?: () => void
  onClick?: (e) => void
  onChange?: (card: ICard) => void
  style?: CSSProperties
  tagStyle?: CSSProperties
  className?: string
  id: string
  index: number
  title?: string
  label?: string
  description?: string
  tags?: TagProps[]
  cardDraggable?: boolean
  editable?: boolean
  t: typeof createTranslate
} & {[key: string]: any}

export const Card: FC<PropsWithChildren<CardProps>> = ({
  onDelete,
  onChange,
  id,
  onClick,
  style,
  className,
  description,
  label,
  t,
  tags,
  title,
  cardDraggable,
  editable,
  showDeleteButton,
  tagStyle
}) => {
  const _onDelete = (e: React.MouseEvent<HTMLDivElement> | React.MouseEvent<StyledComponent<'div', any>>) => {
    onDelete()
    e.stopPropagation()
  }
  const updateCard = (card: Partial<ICard>) => {
    onChange({...card, id})
  }

  return (
    <MovableCardWrapper data-id={id} onClick={onClick} style={style} className={className}>
      <CardHeader>
        <CardTitle draggable={cardDraggable}>
          {editable ? (
            <InlineInput
              value={title}
              border
              placeholder={(t('placeholder.title') as unknown) as string}
              resize="vertical"
              onSave={(value: ICard['title']) => updateCard({title: value})}
            />
          ) : (
            title
          )}
        </CardTitle>
        <CardRightContent>
          {editable ? (
            <InlineInput
              value={label}
              border
              placeholder={(t('placeholder.label') as unknown) as string}
              resize="vertical"
              onSave={(value: ICard['label']) => updateCard({label: value})}
            />
          ) : (
            label
          )}
        </CardRightContent>
        {showDeleteButton && <DeleteButton onClick={_onDelete} />}
      </CardHeader>
      <Detail>
        {editable ? (
          <InlineInput
            value={description}
            border
            placeholder={(t('placeholder.description') as unknown) as string}
            resize="vertical"
            onSave={(value: ICard['description']) => updateCard({description: value})}
          />
        ) : (
          description
        )}
      </Detail>
      {tags &&
        tags.length > 0 && (
          <Footer>
            {tags.map(tag => (
              <Tag key={tag.title} {...tag} tagStyle={tagStyle} />
            ))}
          </Footer>
        )}
    </MovableCardWrapper>
  )
}
