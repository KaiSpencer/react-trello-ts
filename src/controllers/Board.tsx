import React, {createContext, FC, PropsWithChildren} from 'react'
import classNames from 'classnames'
import uuidv1 from 'uuid/v1'
import {BoardContainer} from './BoardContainer'
import * as DefaultComponets from '../components'
import {BoardData} from 'rt/types/Board'
import {store} from 'rt/store/store'

// export class _Board extends Component<{
//   id?: string
//   className?: string
//   components?: typeof DefaultComponets
//   data: BoardData
//   t?: any
// }> {
//   store: any
//   id: any
//   constructor(props) {
//     super(props)
//     this.store = this.getStore()
//     this.id = this.props.id || uuidv1()
//   }

//   getStore = () => {
//     //When you create multiple boards, unique stores are created for isolation
//     return createStore(boardReducer)
//   }

//   render() {
//     const {className, components} = this.props
//     const allClassNames = classNames('react-trello-board', className || '')
//     return (
//       <Provider store={this.store}>
//         <>
//           <components.GlobalStyle />
//           <BoardContainer id={this.id} {...this.props} className={allClassNames} />
//         </>
//       </Provider>
//     )
//   }
// }

export const Board: FC<
  PropsWithChildren<{
    id?: string
    className?: string
    components: typeof DefaultComponets
    data: BoardData
    t?: any
  }>
> = ({data, children, className, components, id, t, ...rest}) => {
  const allClassNames = classNames('react-trello-board', className || '')

  return (
    <BoardContext.Provider value={store}>
      <components.GlobalStyle />
      <BoardContainer
        id={id || uuidv1()}
        data={data}
        components={components}
        t={t}
        className={allClassNames}
        {...rest}
      />
    </BoardContext.Provider>
  )
}

export const BoardContext = createContext(store)
