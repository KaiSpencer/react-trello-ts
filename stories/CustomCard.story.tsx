import { Meta } from "@storybook/react";
import React, { CSSProperties } from "react";
import Board from "../src";
import { CardComponent } from "../src/components/Card";
import { Tag } from "../src/components/Card/Tag";
import { MovableCardWrapper } from "../src/styles/Base";
import { BoardData } from "../src/types/Board";
import { DeleteButton } from "./../src/widgets/DeleteButton";
Board.displayName = "Board";

const CustomCardExample: CardComponent<{
	name: string;
	cardStyle: CSSProperties;
	body: string;
	dueOn: string;
	cardColor: CSSProperties["color"];
	subTitle: string;
	escalationText: string;
}> = ({
	onClick,
	className,
	name,
	cardStyle,
	body,
	dueOn,
	cardColor,
	subTitle,
	tagStyle,
	escalationText,
	tags,
	showDeleteButton,
	onDelete,
	id,
	index,
	t,
}) => {
	const clickDelete = (e) => {
		onDelete();
		e.stopPropagation();
	};

	return (
		<MovableCardWrapper
			onClick={onClick}
			style={cardStyle}
			className={className}
		>
			<header
				style={{
					borderBottom: "1px solid #eee",
					paddingBottom: 6,
					marginBottom: 10,
					display: "flex",
					flexDirection: "row",
					justifyContent: "space-between",
					color: cardColor,
				}}
			>
				<div style={{ fontSize: 14, fontWeight: "bold" }}>{name}</div>
				<div style={{ fontSize: 11 }}>{dueOn}</div>
				{showDeleteButton && <DeleteButton onClick={clickDelete} />}
			</header>
			<div style={{ fontSize: 12, color: "#BD3B36" }}>
				<div style={{ color: "#4C4C4C", fontWeight: "bold" }}>{subTitle}</div>
				<div style={{ padding: "5px 0px" }}>
					<i>{body}</i>
				</div>
				<div
					style={{
						marginTop: 10,
						textAlign: "center",
						color: cardColor,
						fontSize: 15,
						fontWeight: "bold",
					}}
				>
					{escalationText}
				</div>
				{tags && (
					<div
						style={{
							borderTop: "1px solid #eee",
							paddingTop: 6,
							display: "flex",
							justifyContent: "flex-end",
							flexDirection: "row",
							flexWrap: "wrap",
						}}
					>
						{tags.map((tag) => (
							<Tag key={tag.title} {...tag} tagStyle={tagStyle} />
						))}
					</div>
				)}
			</div>
		</MovableCardWrapper>
	);
};

const data: BoardData = {
	lanes: [
		{
			id: "lane1",
			title: "Planned Tasks",
			label: "12/12",
			style: { backgroundColor: "cyan", padding: 20 },
			titleStyle: { fontSize: 20, marginBottom: 15 },
			labelStyle: { color: "#009688", fontWeight: "bold" },
			cards: [
				{
					id: "Card1",
					name: "John Smith",
					dueOn: "due in a day",
					subTitle: "SMS received at 12:13pm today",
					body: "Thanks. Please schedule me for an estimate on Monday.",
					escalationText: "Escalated to OPS-ESCALATIONS!",
					cardColor: "#BD3B36",
					cardStyle: {
						borderRadius: 6,
						boxShadow: "0 0 6px 1px #BD3B36",
						marginBottom: 15,
					},
					metadata: { id: "Card1" },
				},
				{
					id: "Card2",
					name: "Card Weathers",
					dueOn: "due now",
					subTitle: "Email received at 1:14pm",
					body: "Is the estimate free, and can someone call me soon?",
					escalationText: "Escalated to Admin",
					cardColor: "#E08521",
					cardStyle: {
						borderRadius: 6,
						boxShadow: "0 0 6px 1px #E08521",
						marginBottom: 15,
					},
					metadata: { id: "Card1" },
				},
			],
		},
		{
			id: "lane2",
			title: "Long Lane name this is i suppose ha!",
			cards: [
				{
					id: "Card3",
					name: "Michael Caine",
					dueOn: "due in a day",
					subTitle: "Email received at 4:23pm today",
					body:
						"You are welcome. Interested in doing business with you" + " again",
					escalationText: "Escalated to OPS-ESCALATIONS!",
					cardColor: "#BD3B36",
					cardStyle: {
						borderRadius: 6,
						boxShadow: "0 0 6px 1px #BD3B36",
						marginBottom: 15,
					},
					metadata: { id: "Card1" },
					tags: [
						{ title: "Critical", color: "white", bgcolor: "red" },
						{ title: "2d ETA", color: "white", bgcolor: "#0079BF" },
					],
				},
			],
		},
	],
};

export default ({
	title: "Custom Components",
	component: () => {
		return (
			<Board
				tagStyle={{ fontSize: "80%" }}
				data={data}
				draggable={true}
				components={{ Card: CustomCardExample }}
				onCardClick={(cardId, metadata) =>
					alert(
						`Card with id:${cardId} clicked. Has metadata.id: ${metadata.id}`,
					)
				}
			/>
		);
	},
} satisfies Meta<typeof Board>);

export const Card: Meta = {};
