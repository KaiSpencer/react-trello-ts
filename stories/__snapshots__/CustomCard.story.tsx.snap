// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Components Card smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
                 style="background-color: cyan; padding: 20px;"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH"
                  style="font-size: 20px; margin-bottom: 15px;"
            >
              Planned Tasks
            </span>
            <span class="sc-dAlyuH iHiWje">
              <span style="color: rgb(0, 150, 136); font-weight: bold;">
                12/12
              </span>
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                         style="border-radius: 6px; box-shadow: rgb(189, 59, 54) 0px 0px 6px 1px; margin-bottom: 15px;"
                >
                  <header style="border-bottom: 1px solid rgb(238, 238, 238); padding-bottom: 6px; margin-bottom: 10px; display: flex; flex-direction: row; justify-content: space-between; color: rgb(189, 59, 54);">
                    <div style="font-size: 14px; font-weight: bold;">
                      John Smith
                    </div>
                    <div style="font-size: 11px;">
                      due in a day
                    </div>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div style="font-size: 12px; color: rgb(189, 59, 54);">
                    <div style="color: rgb(76, 76, 76); font-weight: bold;">
                      SMS received at 12:13pm today
                    </div>
                    <div style="padding: 5px 0px;">
                      <i>
                        Thanks. Please schedule me for an estimate on Monday.
                      </i>
                    </div>
                    <div style="margin-top: 10px; text-align: center; color: rgb(189, 59, 54); font-size: 15px; font-weight: bold;">
                      Escalated to OPS-ESCALATIONS!
                    </div>
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                         style="border-radius: 6px; box-shadow: rgb(224, 133, 33) 0px 0px 6px 1px; margin-bottom: 15px;"
                >
                  <header style="border-bottom: 1px solid rgb(238, 238, 238); padding-bottom: 6px; margin-bottom: 10px; display: flex; flex-direction: row; justify-content: space-between; color: rgb(224, 133, 33);">
                    <div style="font-size: 14px; font-weight: bold;">
                      Card Weathers
                    </div>
                    <div style="font-size: 11px;">
                      due now
                    </div>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div style="font-size: 12px; color: rgb(189, 59, 54);">
                    <div style="color: rgb(76, 76, 76); font-weight: bold;">
                      Email received at 1:14pm
                    </div>
                    <div style="padding: 5px 0px;">
                      <i>
                        Is the estimate free, and can someone call me soon?
                      </i>
                    </div>
                    <div style="margin-top: 10px; text-align: center; color: rgb(224, 133, 33); font-size: 15px; font-weight: bold;">
                      Escalated to Admin
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Long Lane name this is i suppose ha!
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                         style="border-radius: 6px; box-shadow: rgb(189, 59, 54) 0px 0px 6px 1px; margin-bottom: 15px;"
                >
                  <header style="border-bottom: 1px solid rgb(238, 238, 238); padding-bottom: 6px; margin-bottom: 10px; display: flex; flex-direction: row; justify-content: space-between; color: rgb(189, 59, 54);">
                    <div style="font-size: 14px; font-weight: bold;">
                      Michael Caine
                    </div>
                    <div style="font-size: 11px;">
                      due in a day
                    </div>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div style="font-size: 12px; color: rgb(189, 59, 54);">
                    <div style="color: rgb(76, 76, 76); font-weight: bold;">
                      Email received at 4:23pm today
                    </div>
                    <div style="padding: 5px 0px;">
                      <i>
                        You are welcome. Interested in doing business with you again
                      </i>
                    </div>
                    <div style="margin-top: 10px; text-align: center; color: rgb(189, 59, 54); font-size: 15px; font-weight: bold;">
                      Escalated to OPS-ESCALATIONS!
                    </div>
                    <div style="border-top: 1px solid rgb(238, 238, 238); padding-top: 6px; display: flex; justify-content: flex-end; flex-flow: wrap;">
                      <span class="sc-fPXMVe eCIUDc"
                            style="color: white; background-color: red; font-size: 80%;"
                      >
                        Critical
                      </span>
                      <span class="sc-fPXMVe eCIUDc"
                            style="color: white; background-color: rgb(0, 121, 191); font-size: 80%;"
                      >
                        2d ETA
                      </span>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
